scitasexternal/packages/ansys/package.py

```diff
@@ -69,10 +69,12 @@ def setup_environment(self, spack_env, run_env):
         run_env.prepend_path('PATH', join_path(ansys_prefix, 'autodyn/bin'))
         run_env.prepend_path('PATH', join_path(ansys_prefix, 'fluent/bin'))
         run_env.prepend_path('PATH', join_path(ansys_prefix, 'polyflow/bin'))
-        run_env.prepend_path('PATH', join_path(ansys_prefix, 'tgrid/bin'))
+        if 'version=17.1' in self.spec: run_env.prepend_path('PATH', join_path(ansys_prefix, 'tgrid/bin'))
         run_env.prepend_path('PATH', join_path(ansys_prefix, 'Framework/bin/Linux64'))  # noqa: E501
         run_env.prepend_path('PATH', join_path(ansys_prefix, 'icemcfd/linux64_amd/bin'))  # noqa: E501
         
         run_env.prepend_path('LD_LIBRARY_PATH', join_path(ansys_prefix, 'Framework/bin/Linux64'))  # noqa: E501
-        run_env.prepend_path('LD_LIBRARY_PATH', join_path(ansys_prefix, 'polyflow/polyflow17.1.0/lnamd64/libs'))  # noqa: E501
+        if 'version=17.1' in self.spec: run_env.prepend_path('LD_LIBRARY_PATH', join_path(ansys_prefix, 'polyflow/polyflow17.1.0/lnamd64/libs'))  # noqa: E501
```

Comment:
This is not optimal as it will break again with the next installation of ANSYS.

Could we use the same construct as in line 62 above?

Reply:
Indeed, my first try was like the line 72 for each version as
if 'version=17.1' in self.spec: ...polyflow17.1.0....
if 'version=19.2' in self.spec: ...polyflow19.2.0....
However, the local test showed that the created module didn't contain the line
prepend_path("LD_LIBRARY_PATH", "/ssoft/spack/external/ansys/19.2/v192/polyflow/polyflow19.2.0/lnamd64/libs")
Don't know why?

Reply:
I made the changes. The local test is now ok.

Reply:
Done! Thanks a lot!